packages/devtools_app/lib/src/screens/memory/panes/chart/legend.dart

```diff
@@ -14,6 +14,102 @@ import 'memory_charts.dart';
 import 'memory_events_pane.dart';
 import 'memory_vm_chart.dart';
 
+class MemoryChartWidget extends StatelessWidget {
+  const MemoryChartWidget({
+    super.key,
+    required this.isAndroidVisible,
+    required this.chartController,
+  });
+
+  final bool isAndroidVisible;
+  final MemoryChartPaneController chartController;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+    final legendHeading = theme.hoverTextStyle;
+
+    final legendRows = <Widget>[];
+
+    final events = eventLegendContent(colorScheme.isLight);
+    legendRows.add(
+      Container(
+        padding: legendTitlePadding,
+        child: Text('Events Legend', style: legendHeading),
+      ),
+    );
+
+    var iterator = events.entries.iterator;
+    while (iterator.moveNext()) {
+      final leftEntry = iterator.current;
+      final rightEntry = iterator.moveNext() ? iterator.current : null;
+      legendRows.add(
+        LegendRow(
+          entry1: leftEntry,
+          entry2: rightEntry,
+          chartController: chartController,
+        ),
+      );
+    }
+
+    final vms = vmLegendContent(chartController.vm);
+    legendRows.add(
+      Container(
+        padding: legendTitlePadding,
+        child: Text('Memory Legend', style: legendHeading),
+      ),
+    );
+
+    iterator = vms.entries.iterator;
+    while (iterator.moveNext()) {
+      final legendEntry = iterator.current;
+      legendRows.add(
+        LegendRow(
+          entry1: legendEntry,
+          chartController: chartController,
+        ),
+      );
+    }
+
+    if (isAndroidVisible) {
+      final androids = androidLegendContent(chartController.android);
+      legendRows.add(
+        Container(
+          padding: legendTitlePadding,
+          child: Text('Android Legend', style: legendHeading),
+        ),
+      );
+
+      iterator = androids.entries.iterator;
+      while (iterator.moveNext()) {
+        final legendEntry = iterator.current;
+        legendRows.add(
+          LegendRow(
+            entry1: legendEntry,
+            chartController: chartController,
+          ),
+        );
+      }
+    }
+
+    return Container(
+      width: legendWidth,
+      height: isAndroidVisible ? legendHeight2Charts : legendHeight1Chart,
```

Comment:
we shouldn't need to specify the height here. If should match the height of its content

Reply:
without height the behavior is weird:  

https://user-images.githubusercontent.com/12115586/202821852-ab5aa99a-a2ee-4355-8820-53f85af66256.mov

Reply:
Okay, not a huge deal to leave. But I suspect there were some errors in the console when you were seeing this. Might be an unbounded scrollview error or something like that.

Reply:
Oh can you try to add MainAxisSize.min to the Column below? That should allow you to get rid of the explicit height

Reply:
Did not help:
```
FlutterError (RenderFlex children have non-zero flex but incoming height constraints are unbounded.
When a column is in a parent that does not provide a finite height constraint, for example if it is in a vertical scrollable, it will try to shrink-wrap its children along the vertical axis. Setting a flex on a child (e.g. using Expanded) indicates that the child is to expand to fill the remaining space in the vertical direction.
These two directives are mutually exclusive. If a parent is to shrink-wrap its child, the child cannot simultaneously expand to fit its parent.
Consider setting mainAxisSize to MainAxisSize.min and using FlexFit.loose fits for the flexible children (using Flexible rather than Expanded). This will allow the flexible children to size themselves to less than the infinite remaining space they would otherwise be forced to take, and then will cause the RenderFlex to shrink-wrap the children rather than expanding to fit the maximum constraints provided by the parent.
If this message did not help you determine the problem, consider using debugDumpRenderTree():
  https://flutter.dev/debugging/#rendering-layer
  http://api.flutter.dev/flutter/rendering/debugDumpRenderTree.html
The affected RenderFlex is:
  RenderFlex#721b6 relayoutBoundary=up11 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE(creator: Column ← Padding ← DecoratedBox ← ConstrainedBox ← Container ← MemoryChartLegend ← Padding ← DualValueListenableBuilder<bool, bool> ← Row ← Semantics ← _FocusMarker ← Focus ← ⋯, parentData: offset=Offset(1.0, 5.0) (can use size), constraints: BoxConstraints(w=194.0, 0.0<=h<=Infinity), size: Size(194.0, 194.0), direction: vertical, mainAxisAlignment: start, mainAxisSize: min, crossAxisAlignment: start, textDirection: ltr, verticalDirection: down)
The creator information is set to:
  Column ← Padding ← DecoratedBox ← ConstrainedBox ← Container ← MemoryChartLegend ← Padding ← DualValueListenableBuilder<bool, bool> ← Row ← Semantics ← _FocusMarker ← Focus ← ⋯
The nearest ancestor providing an unbounded width constraint is: RenderFlex#d32b0 relayoutBoundary=up4 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
  needs compositing
  creator: Column-[GlobalKey#7ff88 Chart Hover] ← MemoryBody ← Expanded ← Column ← BannerMessages ← Expanded ← Column ← ValueListenableBuilder<bool> ← AnalyticsPrompt ← _FocusMarker ← Semantics ← FocusScope ← ⋯
  parentData: offset=Offset(0.0, 0.0); flex=1; fit=FlexFit.tight (can use size)
  constraints: BoxConstraints(0.0<=w<=834.0, h=615.0)
  size: Size(834.0, 615.0)
  direction: vertical
  mainAxisAlignment: start
  mainAxisSize: max
  crossAxisAlignment: center
  verticalDirection: down
See also: https://flutter.dev/layout/
If none of the above helps enough to fix this problem, please don't hesitate to file a bug:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md)
```